### PR TITLE
refactor: Some QOL changes to improve DX

### PR DESCRIPTION
**Pull request by akash1810**
> **Note**
> These changes are easiest to review commit by commit.

## What does this change?

Make a few QOL improvements to improve the DX:
  - Use departmental config:
    - Use [departmental tsconfig](https://github.com/guardian/csnx/tree/main/libs/%40guardian/tsconfig) to reduce configuration.
    - Use [departmental linting](https://github.com/guardian/csnx/tree/main/libs/%40guardian/eslint-config-typescript) to improve consistency with projects across the department. 
  - Repository layout[^1]:
    - Move code files to `src` directory, this makes it easier to see code vs config etc.
    - Write compiled output to `dist` directory

## How to test

CI should continue to pass.

## How can we measure success?

Improved consistency with projects across the department.

[^1]: These changes could be made in a separate PR to reduce the diff, as GH is showing the changes as delete and recreate rather than file moves.

### CODE REVIEW

File: src/riffraff.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	stacks: string[];
+	regions: string[];
+	deployments: Record<
+		string,
+		{
+			sources?: string[];
+			[name: string]: unknown;
+		}
+	>;
+};
```
Comment (akash1810) [inline src/riffraff.ts:63]: This has changed from:

```
deployments: { [name: string]: any }
```

To provide more type information for [`index.ts`](https://github.com/guardian/actions-riff-raff/blob/e501b2e28d435b985d6057eee613dad7c24dab4f/src/index.ts#L47-L55).

**Review by nicl — APPROVED** [review]
Looks great! 🚢 

If possible, it would be great to use `git mv` in future to simplify the diff (rather than it thinking everything is deleted/new), though I realise it doesn't always work.

**Comment by akash1810** [comment]
> If possible, it would be great to use `git mv` in future to simplify the diff (rather than it thinking everything is deleted/new), though I realise it doesn't always work.

I had used `git mv` in the initial commit, however as more changes are made in later commits, I think GitHub interprets the net change as delete+new.

**Comment by nicl** [comment]
@akash1810 ah yep thanks! It's the linting that is confusing GH.

**Pull request merged**